api/agent/pure_runner.go

```diff
@@ -612,7 +636,20 @@ func (pr *pureRunner) handleTryCall(tc *runner.TryCall, state *callHandle) error
 		}
 		state.c.slotHashId = string(hashId[:])
 	}
-	pr.spawnSubmit(state)
+	errSubmit := make(chan error, 1)
+	state.c.ackSync = make(chan error, 1)
+	pr.spawnSubmit(state, errSubmit)
+	select {
```

Comment:
this is tricky. we shouldn't block here, since this'll block HandleTryCall(). I think we might need a go-routine here listening on the ack channel.

Reply:
right the handleTryCall should return immediately